File: sdks/ruby/lib/amlexia/health.rb
# frozen_string_literal: true

require "net/http"
require "uri"

module Amlexia
  module Health
    module_function

    def check_ingest_health(ingest_url: Client::DEFAULT_INGEST_URL, timeout: 5)
      uri = URI("#{ingest_url.to_s.chomp('/')}/health")
      start = Process.clock_gettime(Process::CLOCK_MONOTONIC)
      http = Net::HTTP.new(uri.host, uri.port)
      http.use_ssl = uri.scheme == "https"
      http.open_timeout = timeout
      http.read_timeout = timeout
      res = http.get(uri.path)
      latency = ((Process.clock_gettime(Process::CLOCK_MONOTONIC) - start) * 1000).to_i
      { ok: res.is_a?(Net::HTTPSuccess), status: res.code.to_i, latency_ms: latency }
    rescue StandardError
      latency = ((Process.clock_gettime(Process::CLOCK_MONOTONIC) - start) * 1000).to_i
      { ok: false, status: 0, latency_ms: latency }
    end
  end
end

File: sdks/python/src/amlexia/health.py
from __future__ import annotations

import time
from typing import Any, Dict
from urllib import request as urllib_request

DEFAULT_INGEST_URL = "https://ingest.amlexia.com"


def check_ingest_health(ingest_url: str = DEFAULT_INGEST_URL, timeout: float = 5.0) -> Dict[str, Any]:
    base = ingest_url.rstrip("/")
    start = time.time()
    try:
        req = urllib_request.Request(f"{base}/health", method="GET")
        with urllib_request.urlopen(req, timeout=timeout) as resp:
            return {
                "ok": 200 <= resp.status < 300,
                "status": resp.status,
                "latency_ms": int((time.time() - start) * 1000),
            }
    except Exception:
        return {"ok": False, "status": 0, "latency_ms": int((time.time() - start) * 1000)}
